services/brig/src/Brig/IO/Intra.hs

```diff
@@ -154,73 +145,32 @@ onClientEvent orig conn e = do
   -- in the stream.
   push events rcps orig Push.RouteAny conn
 
-updateSearchIndex :: UserId -> UserEvent -> AppIO ()
-updateSearchIndex orig e = case e of
-  -- no-ops
-  UserCreated {} -> return ()
-  UserIdentityUpdated {} -> return ()
-  UserIdentityRemoved {} -> return ()
-  UserLegalHoldDisabled {} -> return ()
-  UserLegalHoldEnabled {} -> return ()
-  LegalHoldClientRequested {} -> return ()
-  UserSuspended {} -> Search.reindex orig
-  UserResumed {} -> Search.reindex orig
-  UserActivated {} -> Search.reindex orig
-  UserDeleted {} -> Search.reindex orig
-  UserUpdated UserUpdatedData {..} -> do
-    let interesting =
-          or
-            [ isJust eupName,
-              isJust eupAccentId,
-              isJust eupHandle
-            ]
-    when (interesting) $ Search.reindex orig
-
-journalEvent :: UserId -> UserEvent -> AppIO ()
-journalEvent orig e = case e of
-  UserActivated acc ->
-    Journal.userActivate acc
-  UserUpdated UserUpdatedData {eupName = Just name} ->
-    Journal.userUpdate orig Nothing Nothing (Just name)
-  UserUpdated UserUpdatedData {eupLocale = Just loc} ->
-    Journal.userUpdate orig Nothing (Just loc) Nothing
-  UserIdentityUpdated (UserIdentityUpdatedData _ (Just em) _) ->
-    Journal.userUpdate orig (Just em) Nothing Nothing
-  UserIdentityRemoved (UserIdentityRemovedData _ (Just em) _) ->
-    Journal.userEmailRemove orig em
-  UserDeleted {} ->
-    Journal.userDelete orig
-  _ ->
-    return ()
+-- (this function shows that things get a lot simpler and type info increases, but i think we
```

Comment:
> this function shows that things get a lot simpler and type info increases

not sure what exactly you mean. What got simpler and where is the additional type info?